core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -132,7 +132,13 @@ private[deploy] class Master(
     webUi.bind()
     masterWebUiUrl = "http://" + masterPublicAddress + ":" + webUi.boundPort
     if (reverseProxy) {
-      masterWebUiUrl = conf.get("spark.ui.reverseProxyUrl", masterWebUiUrl)
+      conf.getOption("spark.ui.reverseProxyUrl") map { reverseProxyUrl =>
+        val proxyUrlNoSlash = reverseProxyUrl.stripSuffix("/")
+        System.setProperty("spark.ui.proxyBase", proxyUrlNoSlash)
+        // If the master URL has a path component, it must end with a slash.
+        // Otherwise the browser generates incorrect relative links
+        masterWebUiUrl = proxyUrlNoSlash + "/"
```

Comment:
This section of code seems like more work than necessary. Can't you just set `masterWebUiUrl = reverseProxyUrl` here and then move `reverseProxyUrl.stripSuffix("/")` into `System.setProperty` making this a two line function?